### PR TITLE
Simplify publish events in agent

### PR DESCRIPTION
**Pull request by colombod**

### CODE REVIEW

**Review by luisquintanilla — APPROVED** [review]
LGTM

File: dotnet/src/Microsoft.AutoGen/Agents/AgentBase.cs [hunk 1]
```diff
@@ -197,6 +197,13 @@ static async ((AgentBase Agent, RpcRequest Request, TaskCompletionSource<RpcResp
         return await completion.Task.ConfigureAwait(false);
     }
 
+    public async ValueTask Publish<T>(T item, string? source = null) where T : IMessage
```
Comment (rysweet) [inline dotnet/src/Microsoft.AutoGen/Agents/AgentBase.cs:200]: this was overdue, thanks. Should it be called PublishMessage()?

File: dotnet/src/Microsoft.AutoGen/Agents/Agents/IOAgent/ConsoleAgent/IHandleConsole.cs [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // IHandleConsole.cs
 
+using Google.Protobuf;
```
Comment (rysweet) [inline dotnet/src/Microsoft.AutoGen/Agents/Agents/IOAgent/ConsoleAgent/IHandleConsole.cs:4]: don't like that this leaks into the handler - is there another way?

Comment (rysweet) [reply dotnet/src/Microsoft.AutoGen/Agents/Agents/IOAgent/ConsoleAgent/IHandleConsole.cs:4]: maybe it doesn't matter, it will always be in the AgentBase

**Review by rysweet — APPROVED** [review]
Preference: going forward I'd love to have all async methods named *Async - should this one be PublishMessageAsync(iMessage)?

File: dotnet/src/Microsoft.AutoGen/Agents/AgentBase.cs (hunk 1 above)
Comment (colombod) [reply dotnet/src/Microsoft.AutoGen/Agents/AgentBase.cs]: done

File: dotnet/src/Microsoft.AutoGen/Agents/Agents/IOAgent/ConsoleAgent/IHandleConsole.cs (hunk 2 above)
Comment (colombod) [reply dotnet/src/Microsoft.AutoGen/Agents/Agents/IOAgent/ConsoleAgent/IHandleConsole.cs:4]: ok

**Pull request merged**